pkg/mimirtool/commands/promql.go

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package commands
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/prometheus/prometheus/promql/parser"
+)
+
+type PromQLCommand struct {
+	query       string
+	prettyPrint bool
+}
+
+func (c *PromQLCommand) Register(app *kingpin.Application, _ EnvVarNames) {
+	promqlCmd := app.Command("promql", "PromQL formatting and editing for Grafana Mimir.")
+
+	// PromQL Format Query Command
+	promqlFormatCmd := promqlCmd.Command("format", "Format PromQL query with Prometheus' string formatter; wrap query in quotes for CLI parsing.").Action(c.formatQuery)
+	promqlFormatCmd.Flag("pretty", "use Prometheus' pretty-print formatter").BoolVar(&c.prettyPrint)
+	promqlFormatCmd.Arg("query", "query to format").Required().StringVar(&c.query)
```

Comment:
[nit] This is an expression, not a query 🙂

Reply:
`promtool` calls it a query https://prometheus.io/docs/prometheus/latest/command-line/promtool/#promtool-promql

Reply:
Consistent terminology :sob:

Let's stick with what Prometheus uses then.